python/sqlflow_submitter/xgboost/dataset.py

```diff
@@ -149,16 +159,15 @@ def thread_worker(slice_id):
         yield xgb.DMatrix('{0}#{0}.cache'.format(dname) if cache else dname)
         return
 
-    for i in range(SLICE_NUM):
+    for i in range(slice_total):
```

Comment:
I believe `slice_total` should be simply defined as
```python
slice_total = SLICE_NUM * nworers
```

Reply:
Thanks for the advice. I left a comment about this.